test/app/views/core/SubscribeModal.spec.coffee

```diff
@@ -128,22 +141,37 @@ describe 'SubscribeModal', ->
       beforeEach ->
         @purchaseRequest.andReturn({status: 200, responseText: '{}'})
 
-      it 'calls hide()', wrapJasmine ->
-        spyOn(@modal, 'hide')
-        yield @modal.onClickSaleButton()
-        expect(@modal.hide).toHaveBeenCalled()
-        expect(@getTrackerEventNames()).toDeepEqual(
-          [ "Started 1 year subscription purchase", "Finished 1 year subscription purchase" ])
+      describe 'when using PayPal', ->
+        it 'calls hide()', wrapJasmine ->
+          spyOn(@modal, 'hide')
+          @modal.onClickSaleButton()
+          yield @payPalButton.click()
+          expect(@modal.hide).toHaveBeenCalled()
+          expect(@getTrackerEventNames()).toDeepEqual(
+            [ "Started 1 year subscription purchase", "Finished 1 year subscription purchase" ])
+
+      describe 'when using Stripe', ->
+        it 'calls hide()', wrapJasmine ->
+          spyOn(@modal, 'hide')
+          @modal.onClickSaleButton()
+          yield @modal.onClickStripeButton()
+          expect(@modal.hide).toHaveBeenCalled()
+          expect(@getTrackerEventNames()).toDeepEqual(
+            [ "Started 1 year subscription purchase", "Finished 1 year subscription purchase" ])
 
     describe 'when the purchase response is 402', ->
       beforeEach ->
         @purchaseRequest.andReturn({status: 402, responseText: '{}'})
+      
+      # NOTE: Not sure what PayPal does for declined cards
```

Comment:
Currently, if there's a PayPal error on the server side, it returns 422. Would be good to handle that, saying simply "there was an error charging PayPal."